### ISSUE TITLE
Setup without Terminal

### ISSUE THREAD

**Issue by mekodak** [nazgul26/PHPRecipebook#149]
How can i setup without using Terminal.

Maybe by using a sql file og with a php install file?

**Comment by nazgul26** [comment]
The SQL for MySQL is under config/schema/schema_MySQL.sql and some sample ingredients are in sample_ingredients.sql

Give that a try.

**Comment by guardian46w** [comment]
For those of us with a hosted website and NO Linux and/or Terminal experience/knowledge, we need a bit more guidance.

Are you saying that if one creates the database and its schema then we can copy all the files to a server folder and PHPRecipebook will work?

**Comment by nazgul26** [comment]
Yes, that would be the first steps.  After you do that you will need to make some changes in the .env file controls the connection to the database you just created.   Make sure you also change the SALT to be something unique.  Also change - 

export ALLOW_PUBLIC_ACCOUNT_CREATION = "false"
to
export ALLOW_PUBLIC_ACCOUNT_CREATION = "true"

So you can make a new account.  Since you didn't use the command line there won't be any accounts will have a password for.  So make a new account and then go into the SQL backend and edit the users table.  Change the access level field for you new user to admin (value = 90).  

Then turn back the .env setting to 

export ALLOW_PUBLIC_ACCOUNT_CREATION = "false"

Sorry it is not easier, using the command line will avoid the extra steps and allow migration to new SQL versions as changes are made.

**Comment by guardian46w** [comment]
Thanks, Todd, but being so unskilled in the ways of the files you talk about, this didn't make much sense to me.  I'm going to attempt the command line method. (At least I can type and hope I don't make any errors.).  But I still have questions about it. In particular this line:
`
export DATABASE_URL="mysql://dbname:password@localhost/phprecipebook?encoding=utf8&timezone=UTC&cacheMetadata=true&quoteIdentifiers=false&persistent=false"
`
I am assuming that the 'dbname' and 'password' in that line is to create a user and password that will be used to login to the phprecipebook webpage after it's created. And that 'localhost/phprecipebook' will be the name of the database the line will create.  If I have assumed incorrectly please let me know.
And, can you tell me what to change to set the timezone to US Central? 

Also in the line:
`export EMAIL_TRANSPORT_DEFAULT_URL="smtp://my@gmail.com:secret@smtp.gmail.com:587?tls=true"
`
Do I change the 'my' to my real Gmail username?

**Comment by nazgul26** [comment]
For the db connection the string there is a bit off, sorry.  

should have been - 

username:password@server/dbname?...

You will need to create the username/password and the database before using the script.

For the email, yes you can put in your own gmail credentials.  It is not essential for the app to run so you can skip that if you like.

**Comment by guardian46w** [comment]
Well, dang it! My hosting plan does not have root access (I bought a cheat one - no sudu). I'll have to do some thinking about it before I decide if its worth the upgrade (the site really is only for my personal amusement).
I really wanted to put this on my online website but I guess I'll try it on my in-the-basement server.

**Comment by guardian46w** [comment]
Hey Todd,
This is the best PHP Recipe script I've found. I had it working at one time but changed Hosts and needed to do a clean install. For us that don't have the tech skills to use Terminal and/or Linux command line and/or don't have Sudo access, it's just too much for us to figure out. Even with Sudo, there are too many steps and commands to get this working. The script needs an easier install method.
At this point I have just given up.

**Comment by withanhdammit** [comment]
What would be super cool is if this was in a Docker container, then it would be super easy for people to install.

**Comment by nazgul26** [comment]
I agree, Docker would probably be a good option.  Never setup a container before... so will give that a try and see how it goes.

**Comment by ilveroluca** [comment]
There was a docker-compose at one point (PR #86)

**Comment by withanhdammit** [comment]
@ilveroluca It looks stale, the last update was almost four years ago.  Any chance you'll refresh that?

**Comment by ilveroluca** [comment]
Never say never, but I don't think I'll find time in the short term. I could support anyone who wants to work on it though.

**Comment by withanhdammit** [comment]
I feel that!  I've never built a docker container before, but near-term at least, there's no room on the clock.  Hopefully @nazgul26 can work it in!

**Comment by nazgul26** [comment]
I am working on it.  Never built a docker image before either.    The PR with the docker files was a good starting place, but not quite right and out of date based on the current code/system.  Should have a MySQL setup working in a couple weeks and then will post out some files to give it a go.

**Comment by nazgul26** [comment]
Release is ready - https://github.com/nazgul26/PHPRecipebook/releases/tag/6.4

Things came together faster then expected.  Give it a go and let me if you hit any issues.

**Comment by guardian46w** [comment]
I'm afraid this doesn't help me much. I have not Idea what Docker is. And if I did, how to install it on my hosting service.
